server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -515,13 +515,15 @@ public void applyClusterState(ClusterChangedEvent event) {
                             // TODO: this catch is bogus, it means the old repo is already closed,
                             // but we have nothing to replace it
                             logger.warn(() -> new ParameterizedMessage("failed to change repository [{}]", repositoryMetadata.name()), ex);
+                            repository = new InvalidRepository(repositoryMetadata, ex);
                         }
                     }
                 } else {
                     try {
                         repository = createRepository(repositoryMetadata, typesRegistry, RepositoriesService::createUnknownTypeRepository);
                     } catch (RepositoryException ex) {
                         logger.warn(() -> new ParameterizedMessage("failed to create repository [{}]", repositoryMetadata.name()), ex);
+                        repository = new InvalidRepository(repositoryMetadata, ex);
                     }
                 }
                 if (repository != null) {
```

Comment:
`repository` cannot be `null` here any more I think.

Reply:
yes, so I remove the `if (repository != null)` and add a asserting here